polyfill/runtest262.mjs

```diff
@@ -0,0 +1,271 @@
+// the entire suite with NODE_V8_COVERAGE, so you should run it in chunks.
+
+// Time the whole thing from start to finish
+const start = process.hrtime.bigint();
+
+// === Utilities and constants ===
+
+function print(str) {
+  process.stdout.write(str + '\n');
+}
+
+// Fancy output only if stdout is a terminal
+color.enabled = process.stdout.isTTY;
+
+// Front matter consists of a YAML document in between /*--- and ---*/
+const frontmatterMatcher = /\/\*---\n(.*)---\*\//ms;
+
+const UTF8 = { encoding: 'utf-8' };
+const GLOB_OPTS = { filesOnly: true };
+
+// === Preparation ===
+
+// Prepare Temporal polyfill. This vm.Script gets executed once for each test,
+// in a fresh VM context.
+
+const polyfillCode = fs.readFileSync('script.js', UTF8);
+const polyfill = new vm.Script(polyfillCode, { filename: path.resolve('script.js') });
+
+// Read the expected failures file and put the paths into a Set
+
+const expectedFailures = new Set(
+  fs
+    .readFileSync('test/expected-failures.txt', UTF8)
+    .split(/\r?\n/g)
+    .filter((line) => line && line[0] !== '#')
+);
+
+// This function reads in a test262 harness helper file, specified in 'includes'
+// in the frontmatter, and caches the resulting vm.Script so it can be used in
+// future tests that also include it.
+
+const helpersCache = new Map();
+function getHelperScript(includeName) {
+  if (helpersCache.has(includeName)) return helpersCache.get(includeName);
+
+  const includeFile = `test262/harness/${includeName}`;
+  const includeCode = fs.readFileSync(includeFile, UTF8);
+  const include = new vm.Script(includeCode);
+
+  helpersCache.set(includeName, include);
+  return include;
+}
+
+// Weed out common error case for people who have just cloned the repo
+if (!fs.statSync('test262/test/').isDirectory()) {
+  print(color.yellow("Missing Test262 directory. Try initializing the submodule with 'git submodule update --init'"));
+  process.exit(1);
+}
+
+const testGlobs = process.argv.slice(2);
+const globResults = testGlobs.flatMap((testGlob) => {
+  let result = globSync(`test262/test/*/Temporal/${testGlob}`, GLOB_OPTS);
+
+  // Fall back to globbing relative to working directory if that didn't match
+  // anything, in case user is using tab completion
+  if (result.length === 0) {
+    result = globSync(testGlob, GLOB_OPTS);
+  }
+
+  result = result.filter((name) => name.endsWith('.js'));
+  if (result.length === 0) {
+    print(color.yellow(`No test files found for pattern: "${testGlob}"`));
+  }
+  return result;
+});
+if (testGlobs.length === 0) {
+  [
+    'test262/test/*/Temporal/**/*.js',
+    // "p*" is a workaround because there is no toTemporalInstant dir at this time
+    'test262/test/built-ins/Date/p*/toTemporalInstant/*.js'
+  ].forEach((defaultGlob) => globResults.push(...globSync(defaultGlob, GLOB_OPTS)));
+}
+
+const testFiles = new Set(globResults);
+const total = testFiles.size;
+if (total === 0) {
+  print('Nothing to do.');
+  process.exit(0);
```

Comment:
Nit: this seems like an error case to me.
```suggestion
  process.exit(1);
```

Reply:
I had this exit with 0 because otherwise the glob `test262/test/built-ins/Date/prototype/toTemporalInstant/*.js` exits with an error while we don't yet have tests for that. But, what I've now done instead, is just not set `failed=1` if that particular invocation of this script fails.